src/edu/ufp/aed2/project/University.java

```diff
@@ -0,0 +1,265 @@
+        // Adding subject to profClass
+        if (!this.subjProf.contains(subject)) {
+            // Subject not registered yet , need to create an ArrayList of professors!
+            ArrayList<Professor> professors = new ArrayList<>();
+            professors.add(professor);
+            this.subjProf.put(subject, professors);
+        } else {
+            // Already contains the subject , just add the professor to ArrayList
+            ArrayList<Professor> professors = this.subjProf.get(subject);
+            if (!professors.contains(professor)) {
+                professors.add(professor);
+            } else {
+                System.out.println("[WARNING] University.java - addProfessorToSubject():");
+                System.out.println("[WARNING] Professor already added to this subject.");
+            }
+        }
+
+    }
+
+
+    /**
+     * Removes a class and its students , rooms , professors from the STs
+     * @param aclass being removed from the university.
+     */
+    public void removeClass(Class aclass){
+        // Get all the data from the class
+        ArrayList<Student> students = aclass.getStudents();
+        Room room = aclass.getSchedule().getRoom();
+        Professor professor = aclass.getProfessor();
+        Subject subject = aclass.getSubject();
+        // Check if the class is in our STs and remove it
+        ArrayList<Class> classes = this.profClass.get(professor);
+        if(classes.contains(aclass)){       // Removes all the data from the STs
+            classes.remove(aclass);         // Remove class from profClass
+            if(classes.isEmpty()) profClass.delete(professor);      // if this professor has no more classes than he's removed
+            ArrayList<Professor> professors = subjProf.get(subject);    // removing professor from the subject taught in this class
+            professors.remove(professor);
+            if(professors.isEmpty()) this.subjProf.delete(subject); // if this subject has no more professor than it's removed
+            ArrayList<Class> classes2 = this.courseClass.get(aclass.getCourse());   // removing class from the the course
+            classes2.remove(aclass);
+            if(classes2.isEmpty()) this.courseClass.delete(aclass.getCourse());
+            ArrayList<Subject> subjects = this.roomSubject.get(room);   // removing subject from the rooms ST
+            subjects.remove(subject);
+            if(subjects.isEmpty()) this.roomSubject.delete(room);       // if this room has no more subjects , delete it
+            for (Student student : students){
+                // removing class for students
+                student.removeClass(aclass);
+            }
+            // removing class for professor
+            professor.removeClass(aclass);
+        } else{
+            System.out.println("[WARNING] University.java - removeClass():");
+            System.out.println("[WARNING] Class not found.");
+        }
```

Comment:
Avoid using `if/else` and use an early return approach. Also, this function is rather complex, consider breaking it into smaller functions.